core/src/main/kotlin/knote/script/PageConfiguration.kt

```diff
@@ -39,61 +55,136 @@ class PageConfiguration : ScriptCompilationConfiguration({
 //                    )
-//
-//                    val sources = importAnnotations.map {
-//                        rootDir.resolve("include").resolve(it.source)
-//                    }.distinct()
-//                    if (sources.isNotEmpty()) {
-//                        importScripts.append(sources.map { it.toScriptSource() })
-//                        reports += ScriptDiagnostic(
-//                            "importScripts += ${sources.map { it.relativeTo(rootDir) }}",
-//                            ScriptDiagnostic.Severity.INFO
-//                        )
-//                    }
+//                    return@onAnnotations ResultWithDiagnostics.Failure(reports)
 //                }
-//            }.asSuccess(reports)
-//        }
+                val pageManager: PageManager = KNote.NOTEBOOK_MANAGER.getPageManager(notebookId) ?: run {
+                    logger.error("pageManager for $notebookId could not be loaded")
+                    reports += ScriptDiagnostic(
+                        "pageManager for $notebookId could not be loaded",
+                        ScriptDiagnostic.Severity.ERROR
+                    )
+                    return@onAnnotations ResultWithDiagnostics.Failure(reports)
+                }
+//                val page = pageManager.findPage(pageId)!! as PageImpl
+                val generatedSrc = rootDir.resolve("build").resolve(".knote").resolve(notebookId).absoluteFile
+                generatedSrc.mkdirs()
+                val pageDependencies = fromPageAnnotations
+                    .map { it.source }
+                    .distinct()
+                    .associate { depId ->
+                        // TODO: DETECT CIRCULAR DEPENDENCIES
+                        if(depId == pageId) {
+                            reports += ScriptDiagnostic(
+                                "page $pageId depends on itself",
+                                ScriptDiagnostic.Severity.FATAL
+                            )
+                            return@onAnnotations ResultWithDiagnostics.Failure(reports)
+                        }
+                        val resultType = pageManager.resultType(depId)
+                        if (resultType == null) {
+                            reports += ScriptDiagnostic(
+                                "resultType of page $depId is unknown",
+                                ScriptDiagnostic.Severity.ERROR
+                            )
+                            return@onAnnotations ResultWithDiagnostics.Failure(reports)
+                        }
+                        depId to resultType
+                    }
+
+                logger.debug("generating dependencies for $notebookId:$pageId")
+                val file = PageDependency.generate(
+                    output = generatedSrc,
+                    notebookId = notebookId,
+                    pageId = pageId,
+                    pageDependencies = pageDependencies
```

Comment:
Note for me: PageDependency is a file and the attributes that make it up

Reply:
thats separate from the GUI, purely for doing the dependency stuff, the GUI should not use it, i think ?